src/reactjs/tests/useQueries.test.tsx

```diff
@@ -776,8 +776,25 @@ describe('useQueries', () => {
       expectType<QueryObserverResult<string, unknown>>(result4[1])
       expectType<QueryObserverResult<number, unknown>>(result4[2])
 
-      // Array as const does not throw error
+      // handles when queryFn returns a Promise
       const result5 = useQueries({
+        queries: [
+          {
+            queryKey: key1,
+            queryFn: () => Promise.resolve('string'),
+            onSuccess: (a: string) => {
+              expectType<string>(a)
```

Comment:
@artysidorenko Out of curiosity, what is the motivation for `expectType` and `expectTypeNotAny` here?

Looking at their [definitions](https://github.com/tannerlinsley/react-query/blob/5562cfaaa04bbcea141c553dbdb211fec8224357/src/reactjs/tests/utils.tsx#L76), they are compile-time checks rather than runtime checks. In this case, the compile-time check is satisfied by `(a: string)` from the previous line in this test. In other words, it's testing the test code rather than the product code.

Am I missing something? I'm not involved in this codebase, just noticed this and thought I'd ask.

Reply:
Hi @symbiont-matthew-pirocchi,

Yeah, the `expectType` and `expectTypeNotAny` are not strictly necessary here, the way the types work at the moment. I think while I was writing the other tests I got into the pattern of test-asserting _every_ type to avoid any surprises.

Am still holding out hope that at some point we'll figure out a way to do the context-sensitive typing on the function parameter, or TS will release a new version that does this more effectively. Then we would replace that snippet with something like this, where the helpers are more important:

```typescript
onSuccess: (a) => {
  expectType<string>(a)
  expectTypeNotAny(a)
},
```

I suppose having these helpers in there already could be a good primer to keep the checks when that happens 🤞

Reply:
@artysidorenko I just stumbled upon eslint-plugin-expect-type, and the PR that adds support for two-slash annotations. Do you think that would be a good replace for type-level tests?
https://twitter.com/joshuakgoldberg/status/1514382066795876352

Reply:
I'm sort of on the fence; the annotation-style looks cool, but reckon you'd still need the same volume of tests to check all the types (just replacing all those `expectType` calls with the two-slashes).